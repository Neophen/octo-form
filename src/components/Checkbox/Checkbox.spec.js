import { shallowMount } from "@vue/test-utils";
import OCheckbox from "./Checkbox";

let wrapper;

describe("OCheckbox", () => {
  beforeEach(() => {
    wrapper = shallowMount(OCheckbox);
  });

  it("is called", () => {
    expect(wrapper.name()).toBe("OCheckbox");
    expect(wrapper.isVueInstance()).toBeTruthy();
  });

  it("render correctly", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("has an input checkbox", () => {
    expect(wrapper.contains("label input[type=checkbox]")).toBeTruthy();
  });

  it("emit input event with value when value change", done => {
    wrapper.setProps({ value: true });
    wrapper.vm.$nextTick(() => {
      expect(wrapper.vm.computedValue).toBeTruthy();
      wrapper.vm.computedValue = false;
      const valueEmitted = wrapper.emitted()["input"][0];
      expect(valueEmitted).toContainEqual(false);
      done();
    });
  });

  it("method focus() gives focus to the input element", done => {
    wrapper.vm.$refs.input.focus = jest.fn();
    wrapper.vm.focus();
    wrapper.vm.$nextTick(() => {
      expect(wrapper.vm.$refs.input.focus).toHaveBeenCalled();
      done();
    });
  });
});
