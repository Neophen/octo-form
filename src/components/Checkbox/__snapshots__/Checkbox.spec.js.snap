// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OCheckbox render correctly 1`] = `
<label class="octo-checkbox"><input type="checkbox" true-value="true" class="octo-checkbox__input" value="false">
  <div class="octo-checkbox__mark">
    <o-icon-stub pack="dashboard" icon="check-mark" size="is-sm" dir="right"></o-icon-stub>
  </div> <span class="octo-checkbox__label"></span>
</label>
`;
